[JIT] Add test for ignored class type property

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45234 [JIT] Add recursive scripting for class type module attributes
* #45262 [JIT] Add API for ignoring arbitrary module attributes
* #45261 [JIT] Enable @unused syntax for ignoring properties
* **#45233 [JIT] Add test for ignored class type property**
* #45232 [JIT] Update docs for recently added features

**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

Differential Revision: [D23971885](https://our.internmc.facebook.com/intern/diff/D23971885)

Commit:
[JIT] Add test for ignored class type property

**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

gmagogsfm commented on test/jit/test_class_type.py:
```diff
@@ -1167,13 +1167,19 @@ def free_function(x: int) -> int:
 
         @torch.jit.script
         class Properties(object):
+            __ignored_properties__ = ["unsupported"]
```
OOC, is `__ignored_properties__` the API we want users to use?

SplitInfinity replied:
I think so. As @ppwwyyxx mentioned it should be easy to enable ignoring properties with a decorator. Should we do it for this release or wait for the next?

eellison replied:
I think it should be good to do for this release - i don't think it's particularly risky (although mabye `unused` is a better fit)

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

eellison reviewed (commented):
One question that was raised by a user internally on your previous PR - should we make the public api `__jit_ignored_properties__` ? or `_jit_ignored_attributes` ? (if we extend this to also work for ignored class attributes which we should) 

Or any other name suggestions?

SplitInfinity commented:
If we go through with D23797598, the public API will be decorator-based. Will we still need this class attribute API for ignoring properties?

eellison commented:
SGTM

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

ydcjeff commented:
> Will we still need this class attribute API for ignoring properties?

Just a little opinion here, for other libraries based on PyTorch, instead of writing decorators everywhere, it is good to have class attribute API for ignoring properties in the respective files in one place.

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

Commit:
Update on "[JIT] Add test for ignored class type property"


**Summary**
This commit modifies `TestClassType.test_properties` to check that
properties on class types can be ignored with the same syntax as
ignoring properties on `Modules`.

**Test Plan**
`python test/test_jit.py TestClassType.test_properties`

[ghstack-poisoned]

eellison commented:
Is this PR still necessary with `__jit_ignored_attributes__` pr above ?

SplitInfinity commented:
> Is this PR still necessary with `__jit_ignored_attributes__` pr above ?

This PR adds the test (which didn't exist at all before and is necessary IMO), and the PR above renames the class property from `__ignored_attributes__` to `__jit_unused_attributes__`. I made this PR before I realized I was going to rename `__ignored_attributes__`. I think they could be merged into one but I don't think it's a big deal.
